synapse/storage/databases/main/event_push_actions.py

```diff
@@ -309,7 +309,45 @@ def add_thread_id_txn(
 
             # Update progress.
             processed_rows = txn.rowcount
-            progress[f"max_{table_name}_stream_ordering"] = max_stream_ordering
+            progress["max_event_push_actions_stream_ordering"] = max_stream_ordering
+            self.db_pool.updates._background_update_progress_txn(
+                txn, "event_push_backfill_thread_id", progress
+            )
+
+            return processed_rows
+
+        def add_thread_id_summary_txn(txn: LoggingTransaction) -> int:
+            min_user_id = progress.get("max_summary_user_id", "")
+            min_room_id = progress.get("max_summary_room_id", "")
+
+            sql = """
+            SELECT user_id, room_id FROM event_push_summary
+            WHERE (user_id, room_id) > (?, ?)
+                AND thread_id IS NULL
```

Comment:
are we expecting many `thread_id`s to be NULL? If not, this query may read a lot of rows

Reply:
Yeah, pretty much everything should be NULL. I was toying with changing it, but note that this is the same behaviour as before.

Reply:
Everything is `NULL` to start with...if we think there isn't that many summaries we can just try to do it all at once, but I suspect this table is quite large.